pkg/auth/authenticator/password/basicauthpassword/basicauthpassword.go

```diff
@@ -49,8 +52,44 @@ type RemoteError struct {
 }
 
 // New returns an authenticator which will make a basic auth call to the given url.
-func New(providerName string, url string, mapper authapi.UserIdentityMapper) authenticator.Password {
-	return &Authenticator{providerName, url, mapper}
+func New(providerName string, url string, client *http.Client, mapper authapi.UserIdentityMapper) authenticator.Password {
```

Comment:
Think we should plumb through the config objects themselves?  It would ensure that we never magic up parameters and help map config to usage for future inspection.  On the other hand, it ties together two not necessarily related layers.  I don't have a strong opinion.